crates/jit/src/instantiate.rs

```diff
@@ -150,8 +151,12 @@ pub fn finish_compile(
         SectionKind::ReadOnlyData,
     );
     let mut total_data_len = 0;
-    for data in data.iter() {
-        obj.append_section_data(data_id, data, 1);
+    for (i, data) in data.iter().enumerate() {
+        // The first data segment gets the requested alignment for the entire
+        // section, but everything afterwards is stored adjacently to the first
+        // so it has an alignment of 1.
```

Comment:
Maybe add a note here why this still works with mmap? In other words why is it OK that we don't align subsequent segments? (Is it that when using mmap-memory mode, all segments are multiple-of-page-sized?)

Reply:
Ah I think this is actually trying to be a bit too clever. Since all the segments are already aligned in their length there's actually no need to pass in a conditional alignment and this can always pass in the desired alignment with an assert that everything ends up adjacent to one another.